fhir-server-test/src/test/java/com/ibm/fhir/server/test/BasicServerTest.java

```diff
@@ -151,13 +229,60 @@ public void testCreatePatientMinimalWithFormat() throws Exception {
-        response = target.path("Patient/" + patientId).request(FHIRMediaType.APPLICATION_FHIR_JSON).header("_format", "application/fhir+json").get();
+        response = target.path("Patient/" + patientId).queryParam("_format", "application/fhir+json").request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
         assertResponse(response, Response.Status.OK.getStatusCode());
         Patient responsePatient = response.readEntity(Patient.class);
 
         TestUtil.assertResourceEquals(patient, responsePatient);
     }
 
+    /**
+     * Create a minimal Patient, then make sure we can retrieve it with varying format with valid FHIR version
+     */
+    @Test(groups = { "server-basic" })
+    public void testCreatePatientMinimalWithFormat_validFhirVersion() throws Exception {
+        WebTarget target = getWebTarget();
+
+        // Build a new Patient and then call the 'create' API.
+        Patient patient = TestUtil.readLocalResource("Patient_DavidOrtiz.json");
+
+        Entity<Patient> entity = Entity.entity(patient, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response = target.path("Patient").request().post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the patient's logical id value.
+        String patientId = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to retrieve the new patient and verify it.
+        response = target.path("Patient/" + patientId).queryParam("_format", "application/fhir+json;fhirVersion=4.0").request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.OK.getStatusCode());
+        Patient responsePatient = response.readEntity(Patient.class);
+
+        TestUtil.assertResourceEquals(patient, responsePatient);
+    }
+
+    /**
+     * Create a minimal Patient, then attempt to retrieve it with varying format with invalid FHIR version
+     */
+    @Test(groups = { "server-basic" })
+    public void testCreatePatientMinimalWithFormat_invalidFhirVersion() throws Exception {
+        WebTarget target = getWebTarget();
+
+        // Build a new Patient and then call the 'create' API.
+        Patient patient = TestUtil.readLocalResource("Patient_DavidOrtiz.json");
+
+        Entity<Patient> entity = Entity.entity(patient, FHIRMediaType.APPLICATION_FHIR_JSON);
+        Response response = target.path("Patient").request().post(entity, Response.class);
+        assertResponse(response, Response.Status.CREATED.getStatusCode());
+
+        // Get the patient's logical id value.
+        String patientId = getLocationLogicalId(response);
+
+        // Next, call the 'read' API to attempt to retrieve the new patient
+        response = target.path("Patient/" + patientId).queryParam("_format", "application/fhir+json;fhirVersion=3.0").request(FHIRMediaType.APPLICATION_FHIR_JSON).get();
+        assertResponse(response, Response.Status.NOT_ACCEPTABLE.getStatusCode());
```

Comment:
makes me think  we should have a lenient mode.

Reply:
Since invalid Content-Type and Accept headers result in HTTP 406/415 as of today, I'm not going to add a lenient mode for this. The `fhirVersion` parameter value is optional.